lib/bandit/http1/handler.ex

```diff
@@ -88,6 +100,50 @@ defmodule Bandit.HTTP1.Handler do
     end
   end
 
+  defp maybe_upgrade_h2c(req, transport_info, method, request_target, headers) do
+    with {:upgrade, "h2c"} <- {:upgrade, Bandit.Headers.get_header(headers, "upgrade")},
+         %Bandit.TransportInfo{secure?: false} <- transport_info,
+         {:ok, connection_headers} <- get_connection_header_keys(headers),
+         {:settings, [{"http2-settings", settings_payload}]} <-
+           {:settings, Enum.filter(headers, fn {key, _value} -> key == "http2-settings" end)},
+         {:ok, settings_payload} <- Base.url_decode64(settings_payload) do
+      Bandit.HTTP1.Adapter.send_resp(
+        req,
+        101,
+        [{"connection", "Upgrade"}, {"upgrade", "h2c"}],
+        <<>>
+      )
+
+      headers =
+        Enum.reject(headers, fn {key, _value} ->
+          key == "connection" || key in connection_headers
+        end)
+
+      {:ok, :h2c, settings_payload, method, request_target, headers}
+    else
+      %Bandit.TransportInfo{secure?: true} -> {:error, "h2c is only supported on http"}
+      {:upgrade, _} -> {:ok, :no_upgrade}
+      {:settings, _} -> {:error, "Expected 1 http2-settings header"}
```

Comment:
Can you reference the source RFC in the error message (RFC123§1.2.3)?

Reply:
Sure. Here it is 5152ee0